cpp/daal/src/threading/threading.cpp

```diff
@@ -247,8 +251,8 @@ DAAL_EXPORT void _daal_del_mutex(void * mutexPtr)
 DAAL_EXPORT bool _daal_is_in_parallel()
 {
 #if defined(__DO_TBB_LAYER__)
-    #if defined(TBB_INTERFACE_VERSION) && TBB_INTERFACE_VERSION >= 12001
-    return tbb::detail::d1::task::current_execute_data() != nullptr;
+    #if defined(TBB_INTERFACE_VERSION) && TBB_INTERFACE_VERSION >= 12002
+    return tbb::task::current_context() != nullptr;
```

Comment:
Shouldn't it be something like:
    #if defined(TBB_INTERFACE_VERSION) && TBB_INTERFACE_VERSION == 12001
    return tbb::detail::d1::task::current_execute_data() != nullptr;
    #elif defined(TBB_INTERFACE_VERSION) && TBB_INTERFACE_VERSION >= 12002
    return tbb::task::current_context() != nullptr;
    #else
    return tbb::task::self().state() == tbb::task::executing;
    #endif

Reply:
No, because there is no TBB with TBB_INTERFACE_VERSION == 12001. It's canceled.